tls/s2n_client_hello.c

```diff
@@ -348,17 +351,20 @@ static S2N_RESULT s2n_client_hello_process_cb_response(struct s2n_connection *co
 
 int s2n_client_hello_recv(struct s2n_connection *conn)
 {
-    POSIX_ENSURE(conn->client_hello.callback_async_blocked == 0, S2N_ERR_ASYNC_BLOCKED);
+    if (conn->client_hello.callback_enable_poll == 0) {
+        POSIX_ENSURE(conn->client_hello.callback_async_blocked == 0, S2N_ERR_ASYNC_BLOCKED);
+    }
 
     if (conn->config->client_hello_cb_mode == S2N_CLIENT_HELLO_CB_BLOCKING ||
```

Comment:
Do you need to check for S2N_CLIENT_HELLO_CB_BLOCKING here? Why is your "parsed" flag not sufficient?